docs/dev-process/coding-guidelines.md

```diff
@@ -180,6 +180,21 @@ See [CODEOWNERS](../../.github/CODEOWNERS) for more information about the area e
   as it obscures the functional changes of the PR.
   A separate PR should be submitted for such refactoring without any functional changes.
 
+* Avoid initializing variables, fields and auto-implemented properties to zero or null because C# guarantees its are initialized to zero.
```

Comment:
I feel this one is not important to be put here. For example, some developers might always make sure all local variables are initialized before use, which is fine and won't actually cause any overhead -- IL code will have the assignment but JIT will be smart enough to not do the assignment.

Instead, I think we should mention the `Interlocked` type here, which I saw in `Managed Threading Best Practices`:
`Consider using the Interlocked class instead of the lock statement to atomically change simple states. The Interlocked class provides better performance for updates that must be atomic.`

Reply:
Will add Interlocked.

We should have a consensus about initializations - it seems your positions is different from the @PaulHigin position.

Reply:
I think `null/zero` initializations for local variables doesn't matter perf-wise. Personally, I prefer to always initialize local variables before use, but that's not something should be enforced to everyone.
For type fields, I saw some are initialized to 0, but rarely see people initialize them to null, but either way, I won't pick it in code review.
For auto properties, I never saw people initialize it to null or zero.

Reply:
I think `null/zero` initializations for local variables doesn't matter perf-wise. Personally, I prefer to always initialize local variables before use, but that's not something should be enforced to everyone.
For type fields, I saw some are initialized to 0, but rarely see people initialize them to null, but either way, I won't pick it in code review.
For auto properties, I never saw people initialize it to null or zero.

Overall, I think we should just remove this item.

Reply:
Removed.